sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -320,8 +301,46 @@ public BlockBlobClient(Uri blobUri, TokenCredential credential, BlobClientOption
         /// New instanc of the <see cref="BlockBlobClient"/> class.
         /// </returns>
         protected static BlockBlobClient CreateClient(Uri blobUri, BlobClientOptions options, HttpPipeline pipeline)
+            => CreateClient(blobUri, options, pipeline, sharedKeyCredential: null);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlockBlobClient"/>
+        /// class.
+        /// </summary>
+        /// <param name="blobUri">
+        /// A <see cref="Uri"/> referencing the block blob that includes the
+        /// name of the account, the name of the container, and the name of
+        /// the blob.
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline
+        /// policies for authentication, retries, etc., that are applied to
+        /// every request.
+        /// </param>
+        /// <param name="pipeline">
+        /// The transport pipeline used to send every request.
+        /// </param>
+        /// <param name="sharedKeyCredential">
+        /// <see cref="StorageSharedKeyCredential"/>
+        /// </param>
+        /// <returns>
+        /// New instanc of the <see cref="BlockBlobClient"/> class.
+        /// </returns>
+        protected static BlockBlobClient CreateClient(
+            Uri blobUri,
+            BlobClientOptions options,
+            HttpPipeline pipeline,
+            StorageSharedKeyCredential sharedKeyCredential)
```

Comment:
This is used by DataLake to initialized `BlockBlobClients`.  Previously, we had a bug where we weren't passing in `StorageSharedKeyCredential`.